.travis.yml

```diff
@@ -38,10 +25,42 @@ before_install:
 install:
   - yarn install --no-lockfile --non-interactive
 
-script:
-  - yarn lint:js
-  # We need to cleanup, since our deploy depends on the git repo's status
-  - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO
+# script:
+#   - yarn lint:js
+#   # We need to cleanup, since our deploy depends on the git repo's status
+#   - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO
+
+notifications:
+  email: false
+
+stages:
+  - test
+  - additional tests
+  - versioned tests
+  - deploy
+
+jobs:
+  fail_fast: true
+
+  include:
+    - stage: test
+      env: NAME=test
+      script: yarn test
+
+    - stage: additional tests
+      env: NAME=floating dependencies
+      install: yarn install --no-lockfile --non-interactive
+      script: yarn test
+
+    - stage: versioned tests
+      env: EMBER_TRY_SCENARIO=ember-lts-2.12
+    - env: EMBER_TRY_SCENARIO=ember-lts-2.16
+    - env: EMBER_TRY_SCENARIO=ember-release
+    - env: EMBER_TRY_SCENARIO=ember-beta
+    - env: EMBER_TRY_SCENARIO=ember-canary
+    - env: EMBER_TRY_SCENARIO=ember-default
 
-after_success:
-  - node_modules/.bin/ember deploy production
+    - stage: deploy
+      if: (branch = master OR tag IS present) AND type = push
```

Comment:
👍 I think this is a lot clearer (and based on the weird off-master deploy today, less error prone) than the current setup.